docs/development.md

```diff
@@ -78,262 +70,33 @@ Then run `M-x add-dir-local-variable RET python-mode RET pyvenv-activate RET <pa
 ((python-mode . ((pyvenv-activate . "~/.cache/pypoetry/virtualenvs/lisa-s7Q404Ij-py3.8"))))
 ```
 
-## Code guideline
-
-### other tools
+### Other setups
 
 - Install and enable [ShellCheck](https://github.com/koalaman/shellcheck) to find bash errors locally.
 
-## Naming Conventions
```

Comment:
All things in red following here were super valid/useful. What's going on?

Reply:
Several parts are removed, not all. The principles,

1. Short files are good files. Many things worth mentioning, such as how to write KISS code, how to create meaningful names, when to create functions... We cannot record all of them.
2. I think that if external contributors want to make important contributions, there will be a lot of discussions and design comments. We can conduct reviews from the beginning, so it relies more on design reviews and code reviews, rather than hoping to read the documentation to make the design better.
3. Focus on LISA specified things. Many tools are used in LISA, but some of them are used by framework only. It's not worth to create document for them. Like #2, if someone want to contribute deeper, we can have more discussions.
4. Common contributions are test cases and extensions. We still need to highlight some parts that are easy to lose but important to LISA, such as how to [write good documents](https://github.com/microsoft/lisa/blob/main/docs/write_case.md).

Reply:
1. Not sure if I follow it. When it comes to documentation, completeness is key, isn't it? I agree that the sections "Naming conventions" (this can be summarized with a link to PEP8), "Committing guidelines" (this is obvious in the good-standards open source world) can go, but I very much like the content of the others. Let's not get rid of something just because it was not committed by us, right?
2. I think it does not hurt to leave that there, it's about good Python, after all. I know there are offenders to those ideas in the code already, and that can serve as a reminder for ourselves to fix later.
3. Do you want to have contributors or not? I don't think there is any value in expecting just test contributions. As a matter of fact, test contributions are the *least* probable we will ever get, as they tend to be specific to user needs.
4. See 3. I disagree.

Reply:
I know some code does not follow right practice in LISA. Feel free to point it out to me, or fix it by PR. I did that several times, and happy to do it continually. We can add some reference if it's needed. Let me know any reference are good ones.

1. For completeness, it may need to copy about 20% content from Fluent Python, git helps and other sources. It must make some assumptions on readers, like they are familiar with programing, and python programing.
2. As our history of LISAv2, RedHat and internal teams contributed a lot of test cases, and we encourage more teams to do that. RedHat team had a big plan on writing test cases but waiting LISAv3. But nobody contributes big improvements to framework. The tests in LISA are important to quality of guest VMs, prevents hundreds of bugs to customers. It's the only value we create LISA.

Reply:
1. I don't mean to include any book-of-Python lenght docs in here, don't get me wrong. It's just that those entries, modulo the ones I mentioned I don't care about, were useful *and* concise enough.
2. Oh, I'll check with other people on the project, then. I thought one the intentions was to get help on the framework as well. Why not?

Reply:
I'm not removing what's not committed by me. The 70% existing content was not created by me. Some guidelines are added since I made the mistakes. I hope the document is in a systematic way and be consistent information level. And I hope to save people's time on reading, it's more friendly to contributors. Some contributors may not be familiar with something, we can discuss in the PR, not ask every contributor to read it. Anyway, I will add a section for extend readings, instead of maintaining content here.

Let me share my thoughts on why removing each part. 
1. Automated Tooling/metadata. It explains how the version is decided. There are many decisions in the code, I didn't see the reason to list this decision here.
2. Automated Tooling/Package Dependencies. It's not reasonable to document used packages separately. The documentation should be in code. And some packages are used in limited scenarios.
3. Automated Tooling/Developer Dependencies. It's renamed to code checks. Some team members cannot setup environment correctly, I hope it helps having manual checks, if it doesn't work. BTW, I didn't get Makefile works on my Windows, which would be worth to document.
4. Type Annotations. It covers in code checks. We forced it if it doesn't work the code cannot be submitted. I didn't see it's difference with other tools like isort, black. I know some people are not familiar with typing in Python, because it's relatively new thing. I will add an extend reading. But learn from code is faster.
5. Runbook schema. Wroten by me, moved to extension document, because it's important to extend schemas.
6. Committing Guidelines, will provide a link. But there should be more document worth to read about how to write comments, API documents... I didn't see the reason to list commit guideline here.
7. Design Patterns. Overall, there are many great points, but most of them are either not practical or related.
    1. "The most important goal we are ... ". That's write, but it's about higher-level designs, not about how to use design patterns.
    2. "We should use caution when using Object Oriented Design ...". I spend half an hour to watch the video, what I learn is, don't write a class with only one method. I don't hope others spent the same time to learn similar things, or even wrong image. Rather than this, I hope to have good readings about object oriented and functional programing.
    3. "This Python Design Patterns is a fantastic collection ...". It's a nice doc, will provide it in extend reading. For people are familiar with design patterns, they just need to map them. For people are not familiar with design patterns, they should learn from what problem the pattern solved. BTW, I don't want to provide like to Gang of Four here, although it's helpful to new programmers. If so, I will provide an endless list.
    4. "Every time a developer chooses to use ...". It's right words, but not design patterns section can solve. The high-level design was reviewed several times and took feedback. There are some internal design documents. and I will provide more. I don't know what else I can do to avoid to "recreate the problems with LISAv2".
    5 "Several popular patterns that actually...". I don't know why the four are considered popular and list here. There are 9 GoF patterns and 4 pythonic patterns in the doc, and I use more on builder, composite patterns. It doesn't make sense either to pick up some or put all of names here.
    6. "Finally, a high-level guide to all things ...". I haven't time to read it. The book says it for both novice and expert. I will put it in extend reading.

Reply:
Alright. We can live with it, let's see. I have to disagree every mention of "learning from code is faster", though:

```
sloccount .

Total Physical Source Lines of Code (SLOC)                = 10,803

```

As for make in Windows, just add this function to your PS profile:
```

function vs_env
{
    $installPath = &"C:\Program Files (x86)\Microsoft Visual Studio\Installer\vswhere.exe" -version 16.0 -property installationpath
    Import-Module (Join-Path $installPath "Common7\Tools\Microsoft.VisualStudio.DevShell.dll")
    Enter-VsDevShell -VsInstallPath $installPath -SkipAutomaticLocation
}
```

or something similar.

Reply:
Learn design from code is impossible but learn how to handle typing is easier. Thank you for hints on getting makefile works. I added a task to enable Makefile instruction. I'm not using PowerShell by default. And it looks need some tools installed, may be visual studio, or c++ tools is enough.

Reply:
Sure. Why don't people use PoweShell by default (on Windows Terminal), when on Windows? It blows my mind that people still accept cmd.exe as a valid/useful thing :(

Reply:
PowerShell is not by default in Windows, and Windows terminal is not default installed. The powershell and cmd are not compatible on quote and some behaviors.

Reply:
I know it's not default, it's just that devs should ditch the default, with is... horrible? :)